### PR TITLE
Improve code coverage for scheduler/api/validation

### PR DESCRIPTION
**Pull request by tanshanshan**
**What this PR does / why we need it**:

Improve code coverage for scheduler/api/validation from #39559

Thanks
**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: plugin/pkg/scheduler/api/validation/validation.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
 )
 
-// Validate checks for errors in the Config
+// ValidatePolicy validate checks for errors in the Config
```
Comment (timothysc) [inline plugin/pkg/scheduler/api/validation/validation.go]: s/validate checks/checks

**Comment by tanshanshan** [comment]
@timothysc ,  PTAL

**Review by timothysc — APPROVED** [review]

File: plugin/pkg/scheduler/api/validation/validation.go [hunk 2]
```diff
@@ -23,7 +23,7 @@ import (
 	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
 )
 
-// Validate checks for errors in the Config
+// ValidatePolicy checks for errors in the Config
 // It does not return early so that it can find as many errors as possible
 func ValidatePolicy(policy schedulerapi.Policy) error {
```
Comment (gyliu513) [inline plugin/pkg/scheduler/api/validation/validation.go]: How about add another test cases as https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/api/validation/validation_test.go#L39-L45

```
TestValidateExtenderWithNonZeroWeight
```

File: plugin/pkg/scheduler/api/validation/validation_test.go [hunk 3]
```diff
@@ -50,3 +50,18 @@ func TestValidatePriorityWithNegativeWeight(t *testing.T) {
 		t.Errorf("Expected error about priority weight not being positive")
 	}
 }
+
+func TestValidateExtenderWithNonZeroWeight(t *testing.T) {
```
Comment (gyliu513) [inline plugin/pkg/scheduler/api/validation/validation_test.go]: s/NonZero/NonNegative

Sorry, I should mention this earlier, I prefer that we use `NonNegative` here as the `Extender` only  have `negative` and `non negative` case.

**Comment by gyliu513** [comment]
lgtm, thanks @tanshanshan

**Comment by tanshanshan** [comment]
@timothysc ,  could you give a lgtm again  please?

Thanks.

**Comment by tanshanshan** [comment]
@timothysc , could you give a approved again please?

Thanks.

**Comment by timothysc** [comment]
@k8s-bot verify test this
@k8s-bot node e2e test this

**Comment by tanshanshan** [comment]
@k8s-bot verify test this

**Comment by timothysc** [comment]
@k8s-bot verify test this

**Comment by timothysc** [comment]
@tanshanshan golint errors - https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/40404/pull-kubernetes-verify/19736/

**Comment by ravisantoshgudimetla** [comment]
ping @tanshanshan. I hope you fixed this. If yes, please request @timstclair for re-running of tests.

**Comment by timothysc** [comment]
@k8s-bot test this #IGNORE

**Comment by timothysc** [comment]
@k8s-bot test this #IGNORE

**Comment by davidopp** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by davidopp** [comment]
Marked as non-release-blocker, but hopefully the test will pass and we can merge it.

**Comment by tanshanshan** [comment]
@timothysc , could you give a lgtm again please?

Thanks.

**Pull request merged**